test/pullrequest/e2e_test.go

```diff
@@ -0,0 +1,170 @@
+package pullrequest
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/jenkins-x/go-scm/scm"
+)
+
+var (
+	proxy = flag.Bool("use-fake-scm-proxy", true, "whether to use fake in-memory proxy instead of using real SCM servers")
+)
+
+func TestPullRequest(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		log.Fatalf("ioutil.TempDir: %v", err)
+	}
+	t.Logf("Test dir: %s", tmpdir)
+	defer os.RemoveAll(tmpdir)
+
+	// Make sure command is installed locally to resolve any dependencies.
+	// We need to do this first since the run might try to pull in new
+	// dependencies through the proxy.
+	install := exec.Command("go", "get", "-u", "github.com/tektoncd/pipeline/cmd/pullrequest-init")
+	t.Log(install)
+	if out, err := install.CombinedOutput(); err != nil {
+		t.Fatal(string(out))
+	}
+
+	for _, tc := range []struct {
+		name string
+		url  string
+
+		// Location of fake SCM data to use for proxy.
+		scmData string
+
+		// Auth token to use.
+		token string
+		// If auth token present, fmt template for Authorization header to check.
+		authHeader string
+	}{
+		{
+			name: "github-default",
+			// Must be http so we can intercept the request in the proxy.
+			url:        "http://github.com/wlynch/test/pulls/1",
+			scmData:    filepath.Join("testdata", "scm", "api.github.com"),
+			token:      envToken("GITHUB_TOKEN"),
+			authHeader: "Bearer %s",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			dir := filepath.Join(tmpdir, tc.name)
+			if err := os.Mkdir(dir, 0755); err != nil {
+				t.Fatalf("os.Mkdir: %v", err)
+			}
+
+			if tc.token != "" {
+				os.Setenv("AUTH_TOKEN", tc.token)
+				defer os.Unsetenv("AUTH_TOKEN")
+			}
+
+			if *proxy {
+				// Stand up fake proxy server. This should be removed if you want to talk to
+				// real SCM providers.
+				h := &handler{
+					Data: tc.scmData,
+					Header: http.Header{
+						"Authorization": []string{fmt.Sprintf(tc.authHeader, tc.token)},
+					},
+				}
+
+				l, err := net.Listen("tcp", "localhost:0")
+				if err != nil {
+					t.Fatalf("net.Listen: %v", err)
+				}
+				t.Logf("starting HTTP proxy on %s", l.Addr().String())
+				go func() {
+					if err := http.Serve(l, h); err != nil {
+						fmt.Println("http.Serve:", err)
+					}
+				}()
+				os.Setenv("http_proxy", l.Addr().String())
+			}
+
+			t.Log("Environment vars:", os.Environ())
+
+			banner(t, "download")
```

Comment:
hmmmm are you sure you want the banner like this? seems like the output from this test is gonna look pretty different from the rest that we have - maybe knowing what it looks like would change my mind.

maybe this is an indication that this test has more output than is needed?